src/Tools/dotnet-monitor/Auth/AuthConfiguratorFactory.cs

```diff
@@ -29,12 +33,34 @@ public static IAuthenticationConfigurator Create(StartupAuthenticationMode start
                     return new MonitorTempKeyAuthConfigurator();
 
                 case StartupAuthenticationMode.Deferred:
-                    // We currently only have one configuration-based authentication mode.
+                    IConfigurationSection authConfigSection = context.Configuration.GetSection(ConfigurationKeys.Authentication);
+                    AuthenticationOptions authOptions = new();
+                    if (authConfigSection.Exists())
+                    {
+                        authConfigSection.Bind(authOptions);
+                        ValidateAuthConfigSection(authOptions, authConfigSection.Path);
+                    }
+
+                    if (authOptions.AzureAd != null)
+                    {
+                        ValidateAuthConfigSection(authOptions.AzureAd, ConfigurationPath.Combine(authConfigSection.Path, ConfigurationKeys.AzureAd));
+                        return new AzureAdAuthConfigurator(authOptions.AzureAd);
+                    }
+
                     return new MonitorKeyAuthConfigurator();
 
                 default:
                     throw new NotSupportedException();
             }
         }
+
+        private static void ValidateAuthConfigSection<T>(T options, string configurationPath)
+        {
+            List<ValidationResult> results = new();
+            if (!Validator.TryValidateObject(options, new ValidationContext(options), results, validateAllProperties: true))
+            {
+                throw new DeferredAuthenticationValidationException(configurationPath, results);
```

Comment:
Because this is thrown during configuration, I think this is going to take down the process. This is not the existing behavior for the api key authentication; it keeps the process going but you have no way to access the api surface because auth is not configured (exception for maybe via Windows auth). This is a valid way to run because automation via collection rules will still work.

Since this only applies to a new feature, it is not technically a breaking change; however, it is a difference in behavior compared to the other authentication options.

Reply:
I'm okay with it as-is, however, if we do end up supporting both API key and AzureAD (and whatever future authN mechanism) side-by-side, then I think this behavior will need to be changed to allow one authN configuration to fail but keep the process running.

We've gone through great lengths to keep dotnet-monitor running (this is the primary reason why configuration of Kestrel is so complicated; the user specifying https without a configured certificate would normally bring down the process but we worked around that in dotnet-monitor; this was done so that a user can setup dotnet-monitor to collect metrics without having to configure the rest of it), even when misconfigured for some aspect. Not having the http API usable, while a big pain, is not catastrophic to dotnet-monitor working for other uses.